LibreNMS/OS/Dlinkap.php

```diff
@@ -38,8 +38,8 @@ public function discoverOS(Device $device): void
         $firmware_oid = $device->sysObjectID . '.5.1.1.0';
         $hardware_oid = $device->sysObjectID . '.5.1.5.0';
 
-        $device->version = snmp_get($device, $firmware_oid, '-Oqv') ?: null;
-        $device->hardware = trim($device->sysDescr . ' ' . snmp_get($device, $hardware_oid, '-Oqv'));
+        $device->version = snmp_get($this->getDeviceArray(), $firmware_oid, '-Oqv') ?: null;
+        $device->hardware = trim($device->sysDescr . ' ' . snmp_get($this->getDeviceArray(), $hardware_oid, '-Oqv'));
```

Comment:
I suspect this doesn't break the tests because we don't have any real test data for it :)